classes/class-settings.php

```diff
@@ -471,6 +471,50 @@ public function get_defaults() {
 		return (array) $defaults;
 	}
 
+	/**
+	 * Retrieves the deletion warning message based on the site type
+	 * and whether or not there is currently a process running to delete the tables.
+	 *
+	 * @return string The deletion warning message.
+	 */
+	public function get_deletion_warning(): string {
+		$site_type = $this->plugin->get_site_type();
+
+		switch ( $site_type ) {
+			case Admin::WP_STREAM_MULTI_NETWORK:
+				$warning = __( 'Warning: This will delete all activity records from the database for all sites.', 'stream' );
+				break;
+			case Admin::WP_STREAM_MULTI_NOT_NETWORK:
+				$warning = $this->plugin->is_large_records_table( Admin::get_blog_record_table_size() ) ?
+						$this->get_async_deletion_warning() :
+						__( 'Warning: This will delete all activity records from the database for this site.', 'stream' );
+				break;
```

Comment:
<!-- highlight specific problems -->
**❗️issue:** I think I found a bug here. I've got an async deletion scheduled and underway. However, on the Settings page I'm getting the latter warning message:

<img width="739" alt="Screenshot 2024-08-06 at 14 36 17" src="https://github.com/user-attachments/assets/39ca4fca-f335-428e-aa92-5e5deda08243">

I think it's because we're checking if it's a large records table based on _current_ table site, i.e.:

```php
$this->plugin->is_large_records_table( Admin::get_blog_record_table_size() )
```

As soon as the async tasks remove bunch of records and the `get_blog_record_table_size()` goes below the threshold set in `is_large_records_table()` method, we won't be seeing the `get_async_deletion_warning()` but rather the other warning message (even though an async tasks are being run in the "background").

Reply:
@delawski YES you're right -- I've changed the logic in https://github.com/xwp/stream/pull/1543/commits/12ca1909854decc9e0326998e4cf884f890ea487 to check if there's an async deletion running first which should avoid this issue.